File: 38-rate-limiting/python-sdk/rate-limit.py
import os, requests
API = "https://api.costkatana.com/api"
KEY = os.getenv("COST_KATANA_API_KEY")

def get_rate_limits():
    res = requests.get(f"{API}/rate-limits",
        headers={"Authorization": f"Bearer {KEY}"})
    res.raise_for_status()
    print(f"⚡ Rate limits: {res.json()['data']}")
    return res.json()["data"]

def update_rate_limits(requests_per_minute, requests_per_hour):
    res = requests.put(f"{API}/rate-limits",
        json={"requestsPerMinute": requests_per_minute, "requestsPerHour": requests_per_hour},
        headers={"Authorization": f"Bearer {KEY}"})
    res.raise_for_status()
    print("✅ Rate limits updated")
    return res.json()["data"]

File: 37-rate-limiting/npm-package/rate-limit.ts
import axios from 'axios';
const API = 'https://api.costkatana.com/api'\;
const KEY = process.env.COST_KATANA_API_KEY;

export async function getRateLimits() {
  const res = await axios.get(`${API}/rate-limits`,
    {headers: {'Authorization': `Bearer ${KEY}`}});
  console.log('⚡ Rate limits:', res.data.data);
  return res.data.data;
}

export async function updateRateLimits(requestsPerMinute: number, requestsPerHour: number) {
  const res = await axios.put(`${API}/rate-limits`, {requestsPerMinute, requestsPerHour},
    {headers: {'Authorization': `Bearer ${KEY}`}});
  console.log('✅ Rate limits updated');
  return res.data.data;
}
